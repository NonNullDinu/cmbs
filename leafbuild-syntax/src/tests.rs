fn parses_without_errors(s: &str) {
    assert_eq!(crate::parser::parse(s).errors, vec![]);
}

// Those are only *parsing* tests, so as long as they make sense
// *syntactically*, they should parse.
// Even if semantically they are void of meaning.
#[test]
fn unit_to_unit_assignment() {
    parses_without_errors("() = ()\n")
}

#[test]
fn line_trailing_space() {
    parses_without_errors("() = () \n")
}

#[test]
fn if_condition() {
    parses_without_errors("if (()) {} else {}")
}

#[test]
fn if_else_if_condition() {
    parses_without_errors("if () {} else if () {} else {}")
}

#[test]
fn if_else_if_else_condition_stretched() {
    parses_without_errors(
        r#"
    if ()

    {



    }

    else if () {


    }

    else {


    }
    "#,
    )
}

#[test]
fn if_condition_in_expr() {
    parses_without_errors("() = if () {} else {} \n")
}

#[test]
fn if_condition_in_expr_stretched() {
    parses_without_errors(
        r#"
    () = if ()
    {

    }

    else
    {


    }
    "#,
    )
}

#[test]
fn declaration() {
    parses_without_errors("let x = ()\n")
}

#[test]
fn foreach_basic() {
    parses_without_errors("foreach () in () {}")
}
