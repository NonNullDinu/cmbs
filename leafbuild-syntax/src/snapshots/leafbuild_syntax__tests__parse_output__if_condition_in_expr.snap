---
source: leafbuild-syntax/src/tests.rs
expression: "() = if () {} else {} \n"

---
- ROOT 0..23
  - Assignment 0..23
    - Expr 0..2
      - PrimaryExpr 0..2
        - TupleExpr 0..2
          - "(" L_PAREN 0..1
          - ")" R_PAREN 1..2
    - " " WHITESPACE 2..3
    - "=" EQ 3..4
    - " " WHITESPACE 4..5
    - Expr 5..21
      - PrimaryExpr 5..21
        - Conditional 5..21
          - ConditionalBranch 5..13
            - "if" IF_KW 5..7
            - " " WHITESPACE 7..8
            - Expr 8..10
              - PrimaryExpr 8..10
                - TupleExpr 8..10
                  - "(" L_PAREN 8..9
                  - ")" R_PAREN 9..10
            - ExprBlock 10..13
              - " " WHITESPACE 10..11
              - "{" L_BRACE 11..12
              - "}" R_BRACE 12..13
          - " " WHITESPACE 13..14
          - "else" ELSE_KW 14..18
          - " " WHITESPACE 18..19
          - ExprBlock 19..21
            - "{" L_BRACE 19..20
            - "}" R_BRACE 20..21
    - " " WHITESPACE 21..22
    - "\n" NEWLINE 22..23

